### PR TITLE
File download and restricted media access

### PR DESCRIPTION
**Pull request by jaredgalanis**
This adds a restricted access field to all media types and the relevant UI changes to display a modal that allows downloading of media attached to a repository item. If restricted access is enabled then a user cannot download the file from the file system and should see no downlink in the download media modal.

To test:
- Run migrations for 10-backend-tests (or add relevant items and media manually)
- Visit repo items and click the download media button
- note the missing download link for media which are flagged as restricted and toggle the restricted flag to see a download link

### CODE REVIEW

File: codebase/config/sync/migrate_plus.migration.idc_ingest_media_remote_video.yml [hunk 1]
```diff
@@ -48,6 +48,8 @@ process:
       bundle: islandora_object
       bundle_key: type
       value_key: field_unique_id
+      value_key: title
```
Comment (bseeger) [inline codebase/config/sync/migrate_plus.migration.idc_ingest_media_remote_video.yml]: I'm guessing this is a typo?  We don't want/need this line.

Comment (jaredgalanis) [reply codebase/config/sync/migrate_plus.migration.idc_ingest_media_remote_video.yml]: yeah maybe it crept in during a prior rebase

**Comment by jabrah** [comment]
* Found that a repository item with a video media has this JS error: 

```
Uncaught Error: Quick Edit could not associate the rendered entity field markup (with [data-quickedit-field-id="media/11/field_media_video_file/en/source"]) with the corresponding rendered entity markup: no parent DOM node found with [data-quickedit-entity-id="media/11"]. This is typically caused by the theme's template for this entity type forgetting to print the attributes.
    processField https://islandora-idc.traefik.me/core/modules/quickedit/js/quickedit.js?v=8.9.17:181
    attach https://islandora-idc.traefik.me/core/modules/quickedit/js/quickedit.js?v=8.9.17:298
    jQuery 2
    attach https://islandora-idc.traefik.me/core/modules/quickedit/js/quickedit.js?v=8.9.17:297
```

* Details page for repository item with a PDF media automatically tries to print the document on page load
* Some reason, the audio player isn't appearing when an audio file as associated with a repo item

**Comment by jabrah** [comment]
Regarding my previous comment:

* I am not able to reproduce the PDF auto-printing issue, so it might have been a weird thing in my local environment
* The quick-edit and audio bullets pointed out above seem to be linked:
  * It looks like both video and audio players do not render unless the the repo item has the "openseadragon" display hint
  * We can address that in a future PR

**Review by jabrah — APPROVED** [review]

**Pull request merged**